### PR TITLE
✨ Run the integration test in CI

### PR DESCRIPTION
**Pull request by MikeSpreitzer**
## Summary
This PR adds a Prow job that runs the integration test.

This is built on #1769 , and will be rebased if that merges first.

## Related issue(s)

Fixes #

### CODE REVIEW

**Comment by MikeSpreitzer** [comment]
The force-push to 715621c8c is the needed rebase.

**Comment by waltforme** [comment]
Not a blocker:
I ran the test manually (as suggested by 'Contributors can run it'), and it was successful. The only thing is that the run ends up with a tarball for etcd as well as the extracted contents of the tarball. Maybe we can do some cleanup at the end? Or check whether we need to download etcd at the beginning.
/lgtm

**Comment by ezrasilvera** [comment]
> Not a blocker: I ran the test manually (as suggested by 'Contributors can run it'), and it was successful. The only thing is that the run ends up with a tarball for etcd as well as the extracted contents of the tarball. Maybe we can do some cleanup at the end? Or check whether we need to download etcd at the beginning. /lgtm

For the CICD it doesn't matter as we delete the VM after the test. However I do prefer we clean up.

File: hack/run-integration-tests.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+if which sha256sum
+then got_sha=$(sha256sum     etcd.tar.gz | awk '{ print $1 }')
+else got_sha=$(shasum -a 256 etcd.tar.gz | awk '{ print $1 }')
+fi
+if [ $expected_sha != "$got_sha" ]; then
+    echo "Got SHA256 $got_sha instead of expected $expected_sha" >& 2
+    exit 1
+fi
+
+tar xzf etcd.tar.gz
```
Comment (ezrasilvera) [inline hack/run-integration-tests.sh:39]: As @waltforme indicated, we can delete the tar at this point

Comment (MikeSpreitzer) [reply hack/run-integration-tests.sh:39]: done.

File: docs/content/v0.20/contributor.md [hunk 2]
```diff
@@ -61,7 +61,7 @@ X ArgoCD CLI
 
 ## Integration testing
 
-There is currently one integration test. Contributors can run it. It is not (yet) run in CI.
+There is currently one integration test. Contributors can run it. There is also a Prow job that runs this test.
```
Comment (ezrasilvera) [inline docs/content/v0.20/contributor.md:64]: This is no longer true (Prow)

Comment (MikeSpreitzer) [reply docs/content/v0.20/contributor.md]: Right. Fixed.

**Pull request merged**